### PR TITLE
Add option `allowGetBody` to allow GET requests with payload

### PR DESCRIPTION
**Pull request by dopecodez**
#### Checklist

- [x] I have read the documentation.
- [x] I have included a pull request description of my changes.
- [x] I have included some tests.
- [x] If it's a new feature, I have included documentation updates.

Fix for #1074

### CODE REVIEW

File: test/arguments.ts [hunk 1]
```diff
@@ -158,10 +158,17 @@ test('throws when passing body with a non payload method', async t => {
 	// @ts-ignore Error tests
 	await t.throwsAsync(got('https://example.com', {body: 'asdf'}), {
 		instanceOf: TypeError,
-		message: 'The `GET` method cannot be used with a body'
+		message: 'The `GET` method cannot be used with a body unless `allowGetBody` option is set to true(Not reccomended)'
```
Comment (szmarczak) [inline test/arguments.ts]:
```suggestion
		message: 'The `GET` method cannot be used with a body'
```

File: readme.md [hunk 2]
```diff
@@ -410,6 +410,15 @@ Default: `true`
 
 By default, redirects will use [method rewriting](https://tools.ietf.org/html/rfc7231#section-6.4). For example, when sending a POST request and receiving a `302`, it will resend the body to the new location using the same HTTP method (`POST` in this case).
 
+###### allowGetBody
+
+Type: `boolean`\
+Default: `false`
+
+By default, GET methods are not allowed to have a payload. This option is to force GET requests to have a payload. Please note that this is an [anti-pattern](https://en.wikipedia.org/wiki/Anti-pattern) and NOT RECOMMENDED. In many servers, the payload in a GET request could be ignored or even lead to errors. Please read the [RF 7321](https://tools.ietf.org/html/rfc7231#section-4.3.1) docs for more information. Please note this option is only enabled for interacting with non-compliant servers when you have no other choice.
```
Comment (szmarczak) [inline readme.md]:
```suggestion
**Note:** The [RFC 7321](https://tools.ietf.org/html/rfc7231#section-4.3.1) doesn't specify any particular behavior for the GET method having a payload, therefore **it's considered an [anti-pattern](https://en.wikipedia.org/wiki/Anti-pattern)**.

Set this to `true` to allow sending body for the `GET` method. However, the [HTTP/2 specification](https://tools.ietf.org/html/rfc7540#section-8.1.3) says that `An HTTP GET request includes request header fields and no payload body`, therefore when using the HTTP/2 protocol this option will have no effect.
```

File: readme.md [hunk 3]
```diff
@@ -410,6 +410,15 @@ Default: `true`
...
+
+**Note:**  **Not Reccomended**
```
Comment (szmarczak) [inline readme.md]:
```suggestion
```

File: source/normalize-arguments.ts [hunk 4]
```diff
@@ -386,6 +388,10 @@ export const normalizeRequestArguments = async (options: NormalizedOptions): Pro
 			throw new TypeError(`The \`${options.method}\` method cannot be used with a body`);
 		}
 
+		if (!options.allowGetBody && (isBody || isForm || isJson) && withoutBodyUnlessSpecified === options.method) {
+			throw new TypeError(`The \`${options.method}\` method cannot be used with a body unless \`allowGetBody\` option is set to true(Not reccomended)`);
```
Comment (szmarczak) [inline source/normalize-arguments.ts]:
```suggestion
			throw new TypeError(`The \`${options.method}\` method cannot be used with a body`);
```

File: readme.md (hunk 2 above)
Comment (sindresorhus) [inline readme.md]: > This option is to force GET requests to have a payload

This makes it sound like Got is actually "forcing" it to have a body. However, the user is still free to not provide one. It just "allows" the user to supply a body.

File: test/arguments.ts [hunk 5]
```diff
@@ -158,10 +158,17 @@ test('throws when passing body with a non payload method', async t => {
...
 	});
 });
 
+test('passes when passing option allowGetBody with a GET method', withServer, async (t, server, got) => {
```
Comment (sindresorhus) [inline test/arguments.ts]:
```suggestion
test('option `allowGetBody` with a GET method', withServer, async (t, server, got) => {
```

File: test/post.ts [hunk 6]
```diff
@@ -20,10 +20,16 @@ const echoHeaders: Handler = (request, response) => {
 	response.end(JSON.stringify(request.headers));
 };
 
-test('GET cannot have body', withServer, async (t, server, got) => {
+test('GET cannot have body without option allowGetBody', withServer, async (t, server, got) => {
```
Comment (sindresorhus) [inline test/post.ts]:
```suggestion
test('GET cannot have body without the `allowGetBody` option', withServer, async (t, server, got) => {
```

File: readme.md (hunk 2 above)
Comment (dopecodez) [reply readme.md]:
```suggestion
By default, GET methods are not allowed to have a payload. This option is to allow GET requests to have a payload. Please note that this is an [anti-pattern](https://en.wikipedia.org/wiki/Anti-pattern) and NOT RECOMMENDED. In many servers, the payload in a GET request could be ignored or even lead to errors. Please read the [RF 7321](https://tools.ietf.org/html/rfc7231#section-4.3.1) docs for more information. Please note this option is only enabled for interacting with non-compliant servers when you have no other choice.
```

Changing the force to allow should make more sense then.

Comment (dopecodez) [reply readme.md]:
```suggestion
By default, GET methods are not allowed to have a payload. This option is to allow GET requests to have a payload. The [RFC 7321](https://tools.ietf.org/html/rfc7231#section-4.3.1) doesn't specify any particular behavior for the GET method having a payload, therefore it's considered an [anti-pattern](https://en.wikipedia.org/wiki/Anti-pattern) and NOT RECOMMENDED. Please note this option is only enabled for interacting with non-compliant servers when you have no other choice.
```

**Comment by sindresorhus** [comment]
Nice work, @dopecodez  👍

**Pull request merged**